compiler/rustc_parse_format/src/lib.rs

```diff
@@ -750,6 +761,29 @@ impl<'a> Parser<'a> {
         }
         if found { Some(cur) } else { None }
     }
+
+    fn suggest_positional_arg_instead_of_captured_arg(&mut self, arg: Argument<'a>) {
+        if let Some(end) = self.consume_pos('.') {
+            let byte_pos = self.to_span_index(end);
+            let start = InnerOffset(byte_pos.0 + 1);
+            let field = self.argument(start);
+            if let ArgumentNamed(_) = arg.position {
+                if let ArgumentNamed(_) = field.position {
+                    self.errors.insert(
+                        0,
+                        ParseError {
+                            description: "field access isn't supported".to_string(),
```

Comment:
Given the check above, this would also trigger if we had something like `"{foo.bar()}"`, right? It'd be interesting to explicitly look for a closing `}`, and if there isn't one, *not* emit a suggestion, just a message with "arbitrary expressions aren't supported" *or* potentially consuming every char until a `}` is found and use that inner span for the suggestion. 🤔 

I feel like the later change can be part of its own PR.

Reply:
I will create an issue about that and work on it later.